cmd/consumer/main.go

```diff
@@ -143,7 +143,7 @@ func appendToFile(filename, text string) error {
 func (rp *RecordProcessor) Shutdown(checkpointer kcl.Checkpointer, reason string) error {
 	if reason == "TERMINATE" {
 		fmt.Fprintf(os.Stderr, "Was told to terminate, will attempt to checkpoint.\n")
-		rp.firehoseWriter.FlushAll()
+		rp.firehoseWriter.Shutdown()
 		rp.checkpoint(checkpointer, "", 0)
```

Comment:
Unfortunately, the underlying batcher doesn't return an error. 

Perhaps we should wait a few seconds here to ensure the background API call completes?
https://github.com/Clever/heka-clever-plugins/blob/master/batcher/message_batcher.go